boards/arm/ti_am62x_sk_m4/doc/index.rst

```diff
@@ -0,0 +1,147 @@
+.. _ti_am62x_sk_m4:
+
+AM62x-SK M4F Core
+#################
+
+Overview
+********
+
+The AM62x-SK board configuration is used by Zephyr applications that run on
+the TI AM62x platform. The board configuration provides support for the ARM
+Cortex-M4F MCU core and the following features:
+
+- Nested Vector Interrupt Controller (NVIC)
+- System Tick System Clock (SYSTICK)
+
+The board configuration also enables support for the semihosting debugging console.
+
+See the `TI AM62X Product Page`_ for details.
+
+.. figure:: img/sk_am62_angled.jpg
+   :align: center
+   :alt: TI AM62x-SK EVM
+
+   Texas Instruments AM62x SK EVM
+
+Hardware
+********
+The AM62x-SK EVM features the AM62x SoC, which is composed of a quad Cortex-A53
+cluster and a single Cortex-M4 core in the MCU domain. Zephyr is ported to run on
+the M4F core and the following listed hardware specifications are used:
+
+- Low-power ARM Cortex-M4F
+- Memory
+
+   - 256KB of SRAM
+   - 2GB of DDR4
+
+- Debug
+
+   - XDS110 based JTAG
+
+Supported Features
+==================
+
+The ti_am62x_sk_m4 configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| SYSTICK   | on-chip    | systick                             |
++-----------+------------+-------------------------------------+
+| PINCTRL   | on-chip    | pinctrl                             |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial                              |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not currently supported by the port.
+
+Devices
+========
+System Clock
+------------
+
+This board configuration uses a system clock frequency of 400 MHz.
+
+DDR RAM
+-------
+
+The board has 2GB of DDR RAM available. This board configuration
+allocates Zephyr 4kB of RAM (only for resource table: 0x9CC00000 to 0x9CC00400).
+
+Serial Port
+-----------
+
+This board configuration uses a single serial communication channel with the
+MCU domain UART (MCU_UART0).
+
```

Comment:
MCU UART0 power domains and clocks are setup by bootloader now right? I think we should mention this somewhere in the documentation.